### PR TITLE
Add recaptcha params to multi option page redirect URI

### PR DESCRIPTION
**Pull request by shanggeeth**
### Proposed changes in this pull request
Add recaptcha  to redirect URI for Multi Option Page from authentication context params, added from basic authenticator

### Related PRs
- https://github.com/wso2-extensions/identity-local-auth-basicauth/pull/155

### CODE REVIEW

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java [hunk 1]
```diff
@@ -897,6 +897,13 @@ protected String getRedirectUrl(HttpServletRequest request, HttpServletResponse
                 }
             }
         }
+        if (StringUtils.isBlank(reCaptchaParamString.toString())) {
+            String captchaParamString = (String) context.getProperty(FrameworkConstants.CAPTCHA_PARAM_STRING);
+            if (captchaParamString != null) {
+                reCaptchaParamString.append(captchaParamString);
```
Comment (Thisara-Welmilla) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java:903]: ReCaptchaParamString, captchaParamString variable names does not indicate the different. Shall we rename them. eg: captchaParamString -> captchaParamStringFromContext.

Comment (shanggeeth) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java]: Addressed in 75cdac0d66fb2eacead4372c14481e4fa6777ba2

**Review by sumedhe — APPROVED** [review]

**Review by dewniMW — APPROVED** [review]

**Comment by dewniMW** [comment]
Only UserDiscoverableApplicationFailureTest and UserDiscoverableApplicationSuccessTest which are known to intermittently fail have been failing in the PR builder. Hence merging the PR.

File: components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java [hunk 2]
```diff
@@ -897,6 +897,14 @@ protected String getRedirectUrl(HttpServletRequest request, HttpServletResponse
                 }
             }
         }
+        if (StringUtils.isBlank(reCaptchaParamString.toString())) {
+            String captchaParamStringFromContext = (String) context.getProperty(
+                    FrameworkConstants.CAPTCHA_PARAM_STRING);
+            if (captchaParamStringFromContext != null) {
+                reCaptchaParamString.append(captchaParamStringFromContext);
```
Comment (Avarjana) [inline components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java:904]: Are we certain that the `captchaParamStringFromContext` is not going to be a blank string? If not, let's go with !isBlank() check to be safe.

Comment (piraveena) [reply components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java:904]: use StringUtils to check for null and blank

**Pull request merged**